Zagrios/bs-manager
# [feat-578] remove dotnet and screen dependency on linux

## Scope

closes https://github.com/Zagrios/bs-manager/issues/578

## Implementation

Removed `dotnet` and `screen` dependencies. Then used `proton` and `wine` binary files which comes package in Steam. They should be linked with the **protonFolder** which is stored in the static config.

On startup as well, there is a check if the protonFolder is correctly configured, else the user will be blocked by a modal to configure the protonFolder. 

## Screenshots

![image](https://github.com/user-attachments/assets/e2027f44-4dbd-42f5-baa7-d7b1d202636c)

![image](https://github.com/user-attachments/assets/78c63902-1af3-4240-bd69-ab34544c23fb)

## How to Test

* Setup
  * Prompting the linux setup modal
    * Remove the **proton-folder** key within `config.json`.

* Feature Tests
  * Check if setting the protonFolder to an invalid value is handled properly, both on the modal and settings page
  * Check if setting the protonFolder to a valid value is correct, both on the modal and settings page

* Functionality Tests
  * Check if you can install mods (especially BSIPA) within anyBSVersion
  * Check if you can launch any BSVersion

## New Resource Text

<details>
<summary>List of added resource text</summary>

**misc.unknown:** An unknown error occurred ¯\\_(ツ)_/¯

**pages.settings.proton-folder.title:** Proton folder
**pages.settings.proton-folder.description:** Change the folder to the Proton path. (eg. Proton - Experimental)
**pages.settings.proton-folder.choose-folder:** Choose folder
**pages.settings.proton-folder.errors.title:** Setting the Proton folder failed
**pages.settings.proton-folder.errors.invalid-folder:** Invalid Proton folder path

**modals.linux-setup.title:** Linux Setup
**modals.linux-setup.proton-folder:** Proton folder
**modals.linux-setup.proton-folder-description:** Choose the folder to the Proton path. (eg. Proton - Experimental)
**modals.linux-setup.choose-folder:** Choose folder

**notifications.bs-launch.errors.PROTON_NOT_SET:** Proton folder not set
**notifications.bs-launch.errors.PROTON_NOT_FOUND:** Proton binary not found
**notifications.bs-launch.msg.PROTON_NOT_SET:** Set the Proton path in settings
**notifications.bs-launch.msg.PROTON_NOT_FOUND:** Set the Proton path in settings

</details>
